lib/validate_partitions.pm

```diff
@@ -0,0 +1,34 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+package validate_partitions;
+
+use base "opensusebasetest";
+
+use strict;
+use warnings;
+use scheduler 'get_test_suite_data';
+use testapi;
+
+my $test_data = get_test_suite_data();
+
+# If we want to validate Msdos partition
+sub validate_partition_table {
+    my $root_disk = script_output("mount |grep \" / \"");
+    my $expected_table_type;
+    if (length($test_data->{table_type})) {
+        $expected_table_type = $test_data->{table_type} }
+    else {
+        $expected_table_type = (check_var('BACKEND', 's390x')) ? "dasd" : "gpt";
+    }
+    my $table_type = script_output("parted -l $root_disk | sed -n -e 4p | awk {'print \$3'}");
```

Comment:
that is something to discuss more deeply but if we are going to do a library, why not parse the command output of `parted -l` in Perl into a hash, so you just need to call $parted->{partition_table} or $parted->{disk}. So we can reuse those commands, check what I did https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/console/validate_encrypt.pm#L28 to inspire you.

Reply:
I considered it, but found it less readable. Maybe because I'm much more comfortable at reading shell than perl... also, requires more lines. Though sure, it would execute faster.

Reply:
I was not thinking in speed, just re-usability, so we will not end up with many implementations to check the same command. But of course this part is up to you how you want it.

Reply:
changed